pkg/controllermanager/controller/shoot/shoot_control_reconcile.go

```diff
@@ -347,6 +347,11 @@ func (c *defaultControl) updateShootStatusReconcileSuccess(o *operation.Operatio
 
 	newShoot, err = kutil.TryUpdateShootStatus(c.k8sGardenClient.Garden(), retry.DefaultRetry, newShoot.ObjectMeta,
 		func(shoot *gardenv1beta1.Shoot) (*gardenv1beta1.Shoot, error) {
+			if shoot.Spec.Kubernetes.KubeProxy == nil {
```

Comment:
Can you factor this logic out into a helper function in pkg/apis/garden/v1beta1? GetKubeProxyMode or sth similar?